### DISCUSSION TITLE
How to change the brand name?

### DISCUSSION THREAD

**Discussion by winsa27** [pglombardo/PasswordPusher#452] [Q&A] [answered]
Hello,

Thank you for this very nice tool.

I installed it on my VPS which runs under centos 7

I first installed docker and then I installed pasword pusher with this command:

docker run -d -p "5100:5100" pglombardo/pwpush-ephemeral:release

I have good access to the interface.

I am now looking to modify the brand.

I read in the documentation that you have to modify the variables in the settings.yml file

When running the command "find -name settings.yml" here is the result:

./9aa903cf5916ae2bc46da3a9cf92ef3cbdda9535be9a04614075ad6ad27af431/diff/opt/PasswordPusher/config/settings.yml
./e3f0409005b9771cf6599e6bc99383724066beaf3da35a4e34f9e02de5c4dfe9/diff/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./a1d86d8fe5cf91d986bb9406ebe7112b065ebd78598ebdc0b2c86281f66bed68/diff/opt/PasswordPusher/config/settings.yml
./9adb38f5bca705a14c3597210523fda2b6e0ce73d903e09ea7a86b54583c90b3/diff/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./11a66b349b8868d1c26d689bf5c43d6b13376d6f156b03f4f8f52672979cfd36/diff/opt/PasswordPusher/config/settings.yml
./34264a798221b310b3e020f788bbcbf18b77ec65923b71038b11382dfbc30e44/diff/opt/PasswordPusher/gemfiles/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./2399769727c8e356aa3b0cbfd8298862a798dc986e7c00604cd11fccf3539fd4/diff/opt/PasswordPusher/config/settings.yml
./22c3cf3f90ba40a86a5b3a167d527afa515e4b6c6887c75886a6fd635856169a/diff/opt/PasswordPusher/gemfiles/mysql/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./66652d5fc9392411e307e2f77e54ae38c787db67f77eb027e0521aa20ec6a844/merged/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./66652d5fc9392411e307e2f77e54ae38c787db67f77eb027e0521aa20ec6a844/merged/opt/PasswordPusher/config/settings.yml

I don't understand why I have several settings.yml files and no matter which one I modify, the modification does not appear.

Can you help me ?

**Comment by winsa27** [comment]
I just ran the following command to remove unused images from docker:

docker system prune --all

When I do find -name settings.yml now here is the result:

./9aa903cf5916ae2bc46da3a9cf92ef3cbdda9535be9a04614075ad6ad27af431/diff/opt/PasswordPusher/config/settings.yml
./e3f0409005b9771cf6599e6bc99383724066beaf3da35a4e34f9e02de5c4dfe9/diff/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml
./66652d5fc9392411e307e2f77e54ae38c787db67f77eb027e0521aa20ec6a844/diff/opt/PasswordPusher/config/settings.yml
./66652d5fc9392411e307e2f77e54ae38c787db67f77eb027e0521aa20ec6a844/merged/opt/PasswordPusher/config/settings.yml
./66652d5fc9392411e307e2f77e54ae38c787db67f77eb027e0521aa20ec6a844/merged/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/config-4.0.0/lib/generators/config/templates/settings.yml

And the result of the following command: docker ps -a

CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES
69c7c42eda80 pglombardo/pwpush-ephemeral:release "bundle exec puma -C…" 9 hours ago Up 9 hours 0.0.0.0:5100->5100/tcp, :::5100->5100/tcp magical_matsumoto
3010f9f7fa9d redis "docker-entrypoint.s…" 11 hours ago Up 9 hours 6379/tcp angry_mcclintock

**Comment by pglombardo** [comment, marked as answer]
Hi @winsa27,

Make sure that you've read [Configuration.md](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md). 

_Currently, the easiest way to configure Password Pusher is by environment variables._

See [the section on rebranding](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#rebranding) for the list of related environment variables..

### Alternative: Configuring with a custom settings.yml file

If you prefer, you can take the [default settings.yml file](https://github.com/pglombardo/PasswordPusher/blob/master/config/settings.yml), modify it and overlay it into the Docker container to apply your customisations.

Inside the [Password Pusher Docker containers](https://hub.docker.com/u/pglombardo) the application code exists in the path `/opt/PasswordPusher/` and the `settings.yml` file is located at `/opt/PasswordPusher/config/settings.yml`.

To replace this file with your own custom version, you can launch the Docker container with a mount option:
`--mount type=bind,source=/path/settings.yml,target=/opt/PasswordPusher/config/settings.yml`

This should explain everything.  If anything still isn't clear, please let me know.
